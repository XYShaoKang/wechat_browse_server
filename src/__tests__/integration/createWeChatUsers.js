import { Prisma, graphqlTestCall } from '../__utils'
import { CreateWeChatUsers } from '../__types'
import { weChatUserData } from '../__variables'

describe('Mutation', () => {
  /** @type {Prisma} */
  let prisma
  beforeEach(() => {
    prisma = new Prisma()
  })
  it('create WeChatUser', async () => {
    const result = await graphqlTestCall({
      query: CreateWeChatUsers,
      variables: weChatUserData,
      context: {
        prisma,
        currentUser: { id: '0' },
        currentWeChat: { id: '0' },
      },
    })
    const id = result.data && result.data.CreateWeChatUsers.id

    expect(id).toBe('0')
    expect(prisma.updateWeChat).toHaveBeenCalled()
    expect(prisma.createWeChatUser).toHaveBeenCalled()
    expect(result).toMatchSnapshot()
    expect(prisma.db.weChatUsers).toMatchSnapshot()
    expect(prisma.db.avatars).toMatchSnapshot()
  })
  it('create WeChatUser Fail No WeChat', async () => {
    const result = await graphqlTestCall({
      query: CreateWeChatUsers,
      variables: weChatUserData,
      context: {
        prisma,
        currentUser: { id: '0' },
      },
    })
    const message = result.errors && result.errors[0].message

    expect(message).toBe('No WeChat')
    expect(result).toMatchSnapshot()
  })
})
