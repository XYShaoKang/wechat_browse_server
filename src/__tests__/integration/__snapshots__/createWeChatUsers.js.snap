// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mutation create WeChatUser 1`] = `
Object {
  "data": Object {
    "CreateWeChatUsers": Object {
      "id": "0",
    },
  },
}
`;

exports[`Mutation create WeChatUser 2`] = `
Array [
  Object {
    "createdAt": "2019-04-21T05:19:31.395Z",
    "id": "0",
    "updatedAt": "2019-04-21T05:19:31.395Z",
    "username": "a",
  },
  Object {
    "createdAt": "2019-04-21T05:19:31.395Z",
    "id": "1",
    "updatedAt": "2019-04-21T05:19:31.395Z",
    "username": "b",
  },
  Object {
    "createdAt": "2019-04-21T05:19:31.395Z",
    "id": "2",
    "updatedAt": "2019-04-21T05:19:31.395Z",
    "username": "c",
  },
  Object {
    "alias": "bbbb",
    "avatar": Object {
      "id": "0",
    },
    "conRemark": "cccc",
    "id": "3",
    "nickname": "dddd",
    "username": "aaaa",
  },
]
`;

exports[`Mutation create WeChatUser 3`] = `
Array [
  Object {
    "bigImg": Object {
      "id": "1",
    },
    "id": "0",
    "thumbnailImg": Object {
      "id": "0",
    },
  },
]
`;

exports[`Mutation create WeChatUser Fail No WeChat 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: No WeChat],
  ],
}
`;
